control/statesp.py

```diff
@@ -1888,3 +1894,62 @@ def ssdata(sys):
     """
     ss = _convert_to_statespace(sys)
     return ss.A, ss.B, ss.C, ss.D
+
+
+def linfnorm(sys, tol=1e-10):
+    """L-infinity norm of a linear system
+
+    Parameters
+    ----------
+    sys : LTI (StateSpace or TransferFunction)
+      system to evalute L-infinity norm of
+    tol : real scalar
+      tolerance on norm estimate
+
+    Returns
+    -------
+    gpeak : non-negative scalar
+      L-infinity norm
+    fpeak : non-negatie scalar
+      Frequency, in rad/s, at which gpeak occurs
+
+    For stable systems, the L-infinity and H-infinity norms are equal;
+    for unstable systems, the H-infinity norm is infinite, while the
+    L-infinity norm is finite if the system has no poles on the
+    imaginary axis.
+
+    For systems with an undefined time base (i.e., sys.dt is None), a
+    ValueError is raised.
```

Comment:
Normally we allow `dt=None` to be treat as either continuous or discrete time, with default being continuous.  Should we do that here?  This is not a major issue since `dt=0` by default, but there are situations where a constant system can be converted to a state space or transfer function with `dt=None` (since it works as either discrete or continuous).

Reply:
I'll drop the ValueError, and use `.isctime()` as suggested below; this will treat these as continuous-time.

One case where `dt=None` by default is static gains (this is from observation, not from reading the code); and, in this case, it doesn't matter what the time base is.